app/models/cart.rb

```diff
@@ -0,0 +1,21 @@
+class Cart < ApplicationRecord
+  has_many :line_items, dependent: :nullify
+
+  def add_item(code, pricing_rules)
+    line_item = line_item_from_code(code)
+    line_item.quantity += 1
+    line_item.update_total(pricing_rules)
+    line_item.save
+  end
```

Comment:
Here we're simply updating the associated line item data based on the selected product (listed as `code`).

I considered encapsulating all of the line item changes within its own class but felt it was unnecessary since logically a cart could just as well be responsible for updating the quantity of its items for example.